cpp/src/arrow/compute/kernels/scalar_validity.cc

```diff
@@ -102,6 +104,15 @@ Status IsNullExec(KernelContext* ctx, const ExecSpan& batch, ExecResult* out) {
     bit_util::SetBitsTo(out_bitmap, out_span->offset, out_span->length, false);
   }
 
+  const auto t = arr.type->id();
+  if (t == Type::SPARSE_UNION) {
+    union_util::SetLogicalSparseUnionNullBits(arr, out_bitmap, out_span->offset);
```

Comment:
I followed the same pattern as we had for `SetNanBits` (used just below, i.e. updating the previously created bitmap), not that this is necessarily the best approach though